File: examples/javascript/apds9002.js
/*
* Author: Zion Orent <zorent@ics.com>
* Copyright (c) 2015 Intel Corporation.
*
* This program and the accompanying materials are made available under the
* terms of the The MIT License which is available at
* https://opensource.org/licenses/MIT.
*
* SPDX-License-Identifier: MIT
*/

// Load Grove Luminance sensor module
var luminanceSensor_lib = require('jsupm_apds9002');

// Instantiate a Grove Luminance sensor on analog pin A0
var myLuminanceSensor_obj = new luminanceSensor_lib.APDS9002(0);

var myInterval = setInterval(function()
{
	console.log("Luminance value is " + myLuminanceSensor_obj.value());
}, 1000);

// On exit: clear interval and memory, print message
process.on('SIGINT', function()
{
	clearInterval(myInterval);
	myLuminanceSensor_obj = null;
	luminanceSensor_lib.cleanUp();
	luminanceSensor_lib = null;
	console.log("Exiting");
	process.exit(0);
});

File: examples/c++/apds9002.cxx
/*
 * Author: Zion Orent <zorent@ics.com>
 * Copyright (c) 2015 Intel Corporation.
 *
 * This program and the accompanying materials are made available under the
 * terms of the The MIT License which is available at
 * https://opensource.org/licenses/MIT.
 *
 * SPDX-License-Identifier: MIT
 */

#include <iostream>
#include <signal.h>

#include "apds9002.hpp"
#include "upm_utilities.h"

using namespace std;

int shouldRun = true;

void
sig_handler(int signo)
{
    if (signo == SIGINT)
        shouldRun = false;
}

int
main()
{
    signal(SIGINT, sig_handler);

    //! [Interesting]
    // Instantiate a Grove Luminance sensor on analog pin A0
    upm::APDS9002 luminance(0);

    while (shouldRun) {
        int val = luminance.value();
        cout << "Luminance value is " << val << endl;

        upm_delay(1);
    }
    //! [Interesting]

    cout << "Exiting" << endl;

    return 0;
}
